### PR TITLE
Initial man pages

### LINKED ISSUE #120: Please add MAN pages to describe the tool
## Please describe what information you were trying to find

If you can describe the information you were trying to find we can structure our documentation better.

...
Currently if I install this tool, I don't have any documentation on the tool at the terminal. We should be documenting the tool as it is built.

### PR DESCRIPTION
**Pull request by engelmi**
This PR addresses #120 and adds the first initial man pages and integrates it into the meson build flow.

### CODE REVIEW

**Comment by engelmi** [comment]
Quote from https://github.com/containers/hirte/pull/153#pullrequestreview-1303717159:
> Why would this be disabled by default? It should be enabled by default. Users of this package should not need to read this document if they want to update the man pages.
> We should have a make docs directive to make this easy to do.

@rhatdan 
You are right. I enabled it now by default. If the necessary tools are missing, the doc build will be skipped and a message will be prompted. 
I am not sure if we need a dedicated make directive as the man pages will be build via `meson install` (or using `make install`) like the other artifacts.

**Comment by rhatdan** [comment]
Could you point me to the source of the man page?  Can the man page be viewed at github.com/containers/hirte?

Usually we build man pages with hirte.md files, and then translate them to man page format with go-md2man.

Then we can view the man page in github.com and on sites like readthecdocs as well as via man command.

Making the man pages easy to contribute to can lead to extra contributors to the project.

**Comment by engelmi** [comment]
The source of the man pages are located in the `man/*.xml`. Oh, I missed adding the generated man pages to the repo - will add them. I used `systemd` and `flatpak` as a reference here, but looking at podman now I see that it uses markdown files for it. Not sure about pros/cons of both as this is the first time I created man pages - and I agree that markdown is more contributer-friendly. 

I can refactor this PR to use .md instead of .xml and use podman as a reference. What do you think? @rhatdan

**Review by mwperina — CHANGES_REQUESTED** [review]
Is it worth to have the content of each man page also in a different format (like HTML in this case)?
If so then having man pages in markdown as source file and then converting them into man page format during the build makes sense. But I'm not sure about go-md2man, we probably don't want to require go environment during the package build so probably pandoc seems like a better alternative to me:

https://eddieantonio.ca/blog/2015/12/18/authoring-manpages-in-markdown-with-pandoc/
https://pandoc.org/

Also please don't forget to include generated man pages into relevant `%files` sections of RPM spec file:

```
%files
...
%{_mandir}/man<N>/<MANPAGE FILE>
...
```

where `<N>` and `<MANPAGE FILE>` need to be replaced with real values

**Comment by rhatdan** [comment]
I find XML difficult impossible to edit, so if a contributor comes along and wants to improve documentation, then editing a markdown file is easy. I don't see where requiring go-md2man is more of a an issue then requireing pandoc.

I find this PR just about impossible to deal with, where just having hirte.md and then having a Makefile directive that launches go-md2man is far simpler.

**Comment by engelmi** [comment]
I mean, I already stated the reasons why XML has initially been chosen and that I'd also prefer markdown. I'll refactor it then and use [go-md2man](https://github.com/cpuguy83/go-md2man). However, as far as I see it there will not only be a hirte.md, but also a hirte-agent.md etc.

**Update:** 
Refactored. Please have another look @rhatdan @mwperina 
Had to disable the markdown linting due to the `% hirte 1` header. Didn't add a make directive since it would be equivalent to a make install. 
@mwperina The rpm spec file change is missing. Is there a way to specify a glob of all files in a directory?

**Comment by rhatdan** [comment]
LGTM, Now it is something I can contribute to.  Sorry for being a pain in the Butt.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -81,6 +86,7 @@ jobs:
                   sed \
                   tar \
               -y
+          go install github.com/cpuguy83/go-md2man/v2@v2.0.2
```
Comment (mwperina) [inline .github/workflows/build.yml]: Could you please install `golang-github-cpuguy83-md2man` RPM from CS9? It's better to align with the version provided in CS9/RHEL9

**Review by mwperina — CHANGES_REQUESTED** [review]

File: hirte.spec.in [hunk 2]
```diff
@@ -9,6 +9,7 @@ Source0:	%{name}-%{version}.tar.gz
 BuildRequires:	gcc
 BuildRequires:	meson
 BuildRequires:	systemd-devel
+BuildRequires:  go-md2man
```
Comment (mwperina) [inline hirte.spec.in]: You need to use proper package name: `golang-github-cpuguy83-md2man`

File: .github/workflows/build.yml (hunk 1 above)
Comment (engelmi) [reply .github/workflows/build.yml]: You are right. I am used to install Go stuff via `go install` and forgot about using dnf.

File: hirte.spec.in [hunk 3]
```diff
@@ -17,7 +18,7 @@ Requires:	systemd
 Hirte is a service orchestrator tool intended for multi-node devices (e.g.: edge devices) clusters with a predefined
 number of nodes and with a focus on highly regulated environment such as those requiring functional safety (for example
 in cars).
-This package contains the orchestrator and command line tool.
+This package contains the orchestrator, node agent and command line tool.
```
Comment (mwperina) [inline hirte.spec.in]: No, node agent is only in hirte-agent RPM

**Review by mwperina — APPROVED** [review]
+1

**Pull request merged**